diskqueue.go

```diff
@@ -460,26 +685,20 @@ func (d *diskQueue) writeOne(data []byte) error {
 	d.writePos += totalBytes
 	d.depth += 1
 
-	fileSize := d.writePos
-
-	if d.diskLimitFeatIsOn {
-		// save space for the number of messages in this file
-		fileSize += 8
-		d.writeBytes += totalBytes
+	if d.enableDiskLimitation {
+		d.totalDiskSpaceUsed += totalBytes
 		d.writeMessages += 1
 	}
 
-	if fileSize >= d.maxBytesPerFile {
+	if reachedFileSizeLimit {
```

Comment:
nit:  reduce diffs between this request and upstream
`if reachedFileSizeLimit` -> `if d.writePos >= d.maxBytesPerFile`